pkg/registry/apis/datasource/utils.go

```diff
@@ -0,0 +1,33 @@
+package datasource
+
+import (
+	"fmt"
+	"strings"
+)
+
+func getInvalidGroupName(pluginId string) string {
+	return "*** InvalidGroupName: " + pluginId + "***" // will not register!
+}
+
+func getDatasourceGroupNameFromPluginID(pluginId string) string {
+	parts := strings.Split(pluginId, "-")
```

Comment:
valid plugin ids may have names like `<org>-<plugin-name-that-can-include-dashes>-<type>`, so maybe we should switch the code below to treat 3+ as the default case, where org is `parts[0]`, type is `parts[len(parts)-1]` and the plugin name itself is s`strings.Join(parts[1:len(parts)-1], "-")`

Reply:
I'll add more tests here too

Reply:
we have a lot of exceptions from this format (especially our own plugins but also community ones).

Reply:
yeah, published plugins should match that scheme but private ones could be anything.

Reply:
is there a problem simply using the whole plugin ID here? Seems unnecessary

Reply:
I don't think we can use pluginId as the group (`/apis/{pluginId}/{version}`) because it _can_ live in the same namespace as any k8s group name -- to play nice in that ecosystem, we are supposed to use domain names that a guaranteed to be unique.  We own/control *.grafana.app

So next option is: `/apis/{pluginId}.grafana.app/{version}` or `/apis/{pluginId}.datasource.grafana.app/{version}` -- that could be OK, but that leads to some pretty embarrassing IDs/URLs:

* /apis/grafana-testdata-datasource.datasource.grafana.app/v0...
* /apis/grafana-testdata.datasource.grafana.app/v0...
* /apis/tempo.grafana.app/v0...

🤷🏻 so I ended up parsing and making them look as clean as possible

Reply:
> `/apis/{pluginId}.grafana.app/{version}` seems like the best option we have.

Agree. This has also additional advantages:
- There's no vagueness from the client perspective what the API group is and by that no need to parse/generate the API group given a plugin ID.
- Plugin ID contains org which is important for creating truly unique API groups per plugin

Reply:
> unfortunately that is not the case already now

I don't understand.  the examples listed all work fine with the proposed approach.

It also means that datasource servers will all have the suffix ".datasource.grafana.app" -- vs needing to look inside and find out.

Private plugins can be anything -- so we don't want them to conflict.

Reply:
if 1:1 mapping for plugin ID is a blocker for folks, i think group == pluginID is maybe the best option.  It has collision issues, and no sure-fire way to know it is a datasource, but seems better than promoting yet another nested "grafana" in the URL 🤣  we would have at least 3 in the URL for cloud!

Reply:
So it's either `/apis/{pluginId}/{version}` or `/apis/{pluginId}.datasource.grafana.app/{version}`. If you want to know it's a datasource by looking at the group you have to pick the latter.

Guess this shouldn't really be a blocker for now since we can end up discussing details forever, we can change it later if needed.

Reply:
yup -- this seems like something that will come up a bunch before this can be in production, so i'll stick with this for now and we can revisit